### PR TITLE
Meshcapade Utilities -> Convert UV

### PR DESCRIPTION
**Pull request by btparker**
In this update:

- Added the Meshcapade Utilities tab
- Added "Convert UV" functionality under the Meshcapade Utilities tab
    - Select source OBJ and FBX for conversion
    - Convert UVs to SMPL v1 (rectified UV coordinates)
- Major refactor to split up monolithic `__init__.py` file
- Changed build/release script
- Placed data files under Git LFS
- Added Flake8 linting

image

Tested on Blender 2.92 and 2.93.

### CODE REVIEW

File: build/make_addon.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+from datetime import date
+
+build_script_dir = path.dirname(path.abspath(__file__))
+today = date.today()
+filedate = today.strftime("%Y%m%d")
+filename = '{}_{}.zip'.format("smplx_blender", filedate)
+
+root_folder = "{}/../".format(build_script_dir)
+addon_folder = "{}/smplx_blender".format(root_folder)
+data_folder = "{}/data".format(addon_folder)
```
Comment (tzaman) [inline build/make_addon.py]: Pathlib makes dealing with folders a little nicer. 

```suggestion
from pathlib import Path

build_script_dir = Path(__file__).parent
today = date.today()
filedate = today.strftime("%Y%m%d")
filename = '{}_{}.zip'.format("smplx_blender", filedate)

root_folder = build_script_dir.parent
addon_folder = root_folder / "smplx_blender"
data_folder = addon_folder / "data"
```

File: smplx_blender/blender.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
+import bpy
+from .globals import (
+    FBX_TYPE,
+    OBJ_TYPE,
+)
+from mathutils import Vector, Quaternion
+
+
+def imported_object(func):
```
Comment (tzaman) [inline smplx_blender/blender.py:9]: Nice!

File: smplx_blender/blender.py [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+    '''Decorator that returns the imported object (Blender is lame and doesn't return from import).
+        Assumes one imported object, and no return from the function
+    '''
+
+    def wrap(*args, **kwargs):
+        active_collection = get_active_collection()
+        old_objs = set(active_collection.objects)
+        func(*args, **kwargs)
+        imported_objs = set(active_collection.objects) - old_objs
+        return imported_objs.pop()
```
Comment (tzaman) [inline smplx_blender/blender.py:21]: Will there always be exactly one?

File: smplx_blender/data/smplx_betas_to_joints_female.json [hunk 4]
```diff
@@ -0,0 +1 @@
..., [0.6816923022270203, 0.014549827203154564, -0.03408398851752281], [0.6983776688575745, 0.015283027663826942, -0.011016267351806164], [0.7181630730628967, 0.009661861695349216, 0.002971678040921688], [-0.74006187915802, 0.021944520995020866, -0.040177930146455765], [-0.7709909081459045, 0.024662749841809273, -0.037520863115787506], [-0.792895495891571, 0.02329767867922783, -0.0371231734752655], [-0.7468854188919067, 0.02426842600107193, -0.06222955882549286], [-0.7762554287910461, 0.025631628930568695, -0.06603360176086426], [-0.7990444898605347, 0.025085289031267166, -0.06971906870603561], [-0.7221466302871704, 0.01678796112537384, -0.09960071742534637], [-0.7364281415939331, 0.016380809247493744, -0.11131169646978378], [-0.7509896755218506, 0.015516774728894234, -0.12266724556684494], [-0.7351065278053284, 0.02168373018503189, -0.08427593111991882], [-0.7618348002433777, 0.023340435698628426, -0.08952361345291138], [-0.7836961150169373, 0.022405201569199562, -0.09650153666734695], [-0.6816904544830322, 0.014549721963703632, -0.03408147022128105], [-0.6983757615089417, 0.015283329412341118, -0.011013735085725784], [-0.718163013458252, 0.009662178345024586, 0.002972340676933527]]}
```
Comment (tzaman) [inline smplx_blender/data/smplx_betas_to_joints_female.json:1]: Should / can these be NPY files? Storing floats in JSON for use in calculations seems hinky from both a precision and size perspective. 

Similarly -- what's in the PKLs and can they be NPYs instead? Pickle as a storage format has issues.

**Review by tzaman — APPROVED** [review]
LGTM! Some minor suggestions and questions about file formats. 
I didn't really look closely at the blender props / UI stuff.

File: smplx_blender/blender.py (hunk 3 above)
Comment (btparker) [reply smplx_blender/blender.py:21]: We'll only support one at any rate, I'll add a comment to that effect though.

File: smplx_blender/data/smplx_betas_to_joints_female.json (hunk 4 above)
Comment (btparker) [reply smplx_blender/data/smplx_betas_to_joints_female.json:1]: I'm fine with converting, I was just leaving previous functionality alone (only refactoring, not changing). I'll do so in a separate PR!

**Pull request merged**